chain/network/src/config.rs

```diff
@@ -101,57 +104,82 @@ pub struct NetworkConfig {
     pub outbound_disabled: bool,
     /// Whether this is an archival node.
     pub archive: bool,
+    /// Feature flag: true if PeerManagerActor should maintain accounts_data::Cache and
+    /// exchange SyncAccountsData with peers. See accounts_data/mod.rs.
+    pub enable_accounts_data: bool,
 }
 
 impl NetworkConfig {
+    // Constructs and validates the config.
+    // TODO(gprusak): make the output immutable, to enforce the invariants
+    // checked during validation: either make it return an Arc, or add an Inner type,
+    // so that NetworkConfig dereferences to Inner.
     pub fn new(
         cfg: crate::config_json::Config,
         node_key: SecretKey,
         validator_signer: Option<Arc<dyn ValidatorSigner>>,
         archive: bool,
-    ) -> Self {
-        Self {
+        enable_accounts_data: bool,
```

Comment:
Not a huge fan of bool arguments. My preferred pattern here would be for the caller to do: 

```
let mut config = Config::new(...);
config.archive = should_be_archive();
if is_sharnet {
  config.enable_acconuts_data = true;
}
```

That is, don't stuff everything into single new call, make stuff more composable and builder-y, as long as you don't spend thousands lines of code on this.